File: shared/setup/sandbox/Dockerfile.java
# Dockerfile for Java Development
# 使用 Maven 构建工具

FROM eclipse-temurin:17-jdk-alpine

# 安装系统依赖
RUN apk add --no-cache \
    git \
    curl \
    bash \
    maven

# 可选：安装 jenv（用于多版本管理）
RUN curl -L -s get.jenv.io | bash
ENV PATH="/root/.jenv/bin:${PATH}"
RUN echo 'eval "$(jenv init -)"' >> /root/.bashrc

# 设置工作目录
WORKDIR /workspace

# 复制 Maven 配置
COPY pom.xml ./

# 下载依赖（利用 Docker 缓存层）
RUN mvn dependency:go-offline -B

# 默认命令
CMD ["/bin/bash"]

File: en/shared/setup/sandbox/Dockerfile.go
# Dockerfile for Go Development

FROM golang:1.21-alpine

# 安装系统依赖
RUN apk add --no-cache \
    git \
    curl \
    bash \
    make \
    gcc \
    musl-dev

# 安装 golangci-lint
RUN curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b /usr/local/bin v1.55.2

# 可选：安装 goenv
RUN git clone https://github.com/go-nv/goenv.git ~/.goenv
ENV PATH="/root/.goenv/bin:${PATH}"
RUN echo 'eval "$(goenv init -)"' >> /root/.bashrc

# 设置 Go 环境变量
ENV GOPATH=/go
ENV PATH="${GOPATH}/bin:/usr/local/go/bin:${PATH}"

# 设置工作目录
WORKDIR /workspace

# 复制 go.mod 和 go.sum
COPY go.mod go.sum ./

# 下载依赖
RUN go mod download

# 默认命令
CMD ["/bin/bash"]
